torch/_inductor/lowering.py

```diff
@@ -1313,8 +1314,14 @@ def arange(
     if isinstance(step, float) and int(step) == step:
         step = int(step)
 
+    def any_arg(typ):
+        args = (start, end, step)
+        return any(isinstance(x, typ) for x in args)
+
     # Triton kernel doesn't support float arange yet, fallback to aten.arange
-    if not (isinstance(start, int) and isinstance(end, int) and isinstance(step, int)):
+    if any_arg(float) and any_arg(sympy.Expr):
+        raise NotImplementedError("Fallback with float and sympy.Expr")
+    elif any_arg(float):
```

Comment:
workaround to avoid going to the fallback

Reply:
note how this still goes here if it's a float. see the comment above about triton not supporting float arange. but i don't know whether it actually works or not.